client/kv.go

```diff
@@ -0,0 +1,168 @@
+func newKvClient(name string, curpClient curpClient, token string) kvClient {
+	return kvClient{name: name, curpClient: curpClient, token: token}
+}
+
+// Get a range of keys from the store
+func (c kvClient) Range(request *xlineapi.RangeRequest) (*xlineapi.RangeResponse, error) {
+	keyRange := []*xlineapi.KeyRange{
+		{
+			Key:      request.Key,
+			RangeEnd: request.RangeEnd,
+		},
+	}
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_RangeRequest{
+			RangeRequest: request,
+		},
+	}
+	proposeId := c.generateProposeId()
+	cmd := xlineapi.Command{Keys: keyRange, Request: &requestWithToken, ProposeId: proposeId}
+	fmt.Println("temp")
+	res, err := c.curpClient.propose(&cmd, true)
+	fmt.Println(res, err)
+	if err != nil {
+		return nil, err
+	} else {
+		return res.CommandResp.GetRangeResponse(), err
+	}
+}
+
+// Put a key-value into the store
+func (c kvClient) Put(request *xlineapi.PutRequest) (*xlineapi.PutResponse, error) {
+	keyRange := []*xlineapi.KeyRange{
+		{
+			Key:      request.Key,
+			RangeEnd: request.Key,
+		},
+	}
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_PutRequest{
+			PutRequest: request,
+		},
+	}
+	proposeId := c.generateProposeId()
+	cmd := xlineapi.Command{Keys: keyRange, Request: &requestWithToken, ProposeId: proposeId}
+	res, err := c.curpClient.propose(&cmd, true)
+	if err != nil {
+		return nil, err
+	} else {
+		return res.CommandResp.GetPutResponse(), err
+	}
+}
+
+// Delete a range of keys from the store
+func (c kvClient) Delete(request *xlineapi.DeleteRangeRequest) (*xlineapi.DeleteRangeResponse, error) {
+	keyRange := []*xlineapi.KeyRange{
+		{
+			Key:      request.Key,
+			RangeEnd: request.RangeEnd,
+		},
+	}
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_DeleteRangeRequest{
+			DeleteRangeRequest: request,
+		},
+	}
+	proposeId := c.generateProposeId()
+	cmd := xlineapi.Command{Keys: keyRange, Request: &requestWithToken, ProposeId: proposeId}
+	res, err := c.curpClient.propose(&cmd, true)
+	if err != nil {
+		return nil, err
+	} else {
+		return res.CommandResp.GetDeleteRangeResponse(), err
+	}
+}
+
+// Creates a transaction, which can provide serializable writes
+func (c kvClient) Txn(request *xlineapi.TxnRequest) (*xlineapi.TxnResponse, error) {
+	var keyRange []*xlineapi.KeyRange
+
+	for _, cmp := range request.Compare {
+		r := xlineapi.KeyRange{
+			Key:      cmp.Key,
+			RangeEnd: cmp.RangeEnd,
+		}
+		keyRange = append(keyRange, &r)
+	}
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_TxnRequest{
+			TxnRequest: request,
+		},
+	}
+	proposeId := c.generateProposeId()
+	cmd := xlineapi.Command{Keys: keyRange, Request: &requestWithToken, ProposeId: proposeId}
+
+	res, err := c.curpClient.propose(&cmd, false)
```

Comment:
Please check err returned by propose first

Reply:
Thank you for the feedback. I've updated the pull request based on your suggestions.